lake-primitives/src/types/block.rs

```diff
@@ -121,14 +119,17 @@ impl Block {
         &self.state_changes
     }
 
-    pub fn action_by_receipt_id(
+    pub fn actions_by_receipt_id(
         &mut self,
         receipt_id: &super::ReceiptId,
-    ) -> Option<&receipts::Action> {
+    ) -> Vec<&(impl Action + std::fmt::Debug + Clone)> {
```

Comment:
I would say the next move of most developers would be to filter it to narrow it down to something they need. So I am thinking about returning an `Iter` here.

Those who are done would just call `.collect`

```
let actions = block
    .actions_by_receipt_id(&receipt_id)
    .collect()
```

Others would be able to proceed with `.filter` or `.map`

```
let function_calls_in_receipt = block
    .actions_by_receipt_id(&receipt_id)
    .filter_map(|action| action.as_function_call())

...
```

Reply:
Yes, that makes perfect sense, and the same is applicable to the `.events` method